hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsOutputStream.java

```diff
@@ -144,6 +151,7 @@ public AbfsOutputStream(
     this.completionService = new ExecutorCompletionService<>(this.threadExecutor);
     this.cachedSasToken = new CachedSASToken(
         abfsOutputStreamContext.getSasTokenRenewPeriodForStreamsInSeconds());
+    ioStatistics = outputStreamStatistics.getIOStatistics();
```

Comment:
1. make a `this.` for consistency
2. Is there ever a risk that outputStreamStatistics is null/has empty iostats at this point?

Reply:
the way we currently create files in AzureblobFileSystemStore, we always give outputStreamStatistics in the context builder, but if we explicitly build an outputStream without outputStream stats, then we could have null pointer exceptions. If we are to have those checks then I'll have to branch a few methods using IOStatistics and outputStream stats to avoid this too.